src/utils/work/seekWorkResponse.js

```diff
@@ -5,34 +5,43 @@ import fetchAPI from 'utils/fetchAPI';
 import unpackResult from 'utils/work/unpackResult';
 import WorkResponseError from 'utils/WorkResponseError';
 
+const throwResponseError = (response) => {
+  throw new Error(`Error ${response.status}: ${response.text}`, { cause: response });
+};
+
 const seekFromS3 = async (ETag, experimentId) => {
   const response = await fetchAPI(`/v1/workResults/${experimentId}/${ETag}`);
-  const { signedUrl } = await response.json();
 
-  if (!signedUrl) return null;
+  if (!response.ok) {
+    if (response.status === 404) return null;
+
+    throwResponseError(response);
+  }
+
+  const { signedUrl } = await response.json();
   const storageResp = await fetch(signedUrl);
+
   if (!storageResp.ok) {
-    return null;
+    throwResponseError(storageResp);
   }
 
   return unpackResult(storageResp);
 };
 
-const seekFromAPI = async (
```

Comment:
why renaming to `dispatchWorkRequest`? `seekFromAPI` sounds clearer to me

Reply:
Because we do not 'seek the result from the API' anymore, as the function does not return the work result anymore. Rather, it's a 'here's the work request, and let me know if it's done'.

Reply:
I get the point, it definitely isn't seeking from the API. I agree with the name change, sounds more accurate